File: mpp-proiect-repository-cristinahendre/lab9(Java)/Java/lab6/modul/DomainModule/src/main/java/Domain/Voluntar.java
package Domain;

import java.io.Serializable;
import java.util.Objects;

public class Voluntar  implements Serializable {


    public Voluntar(){}
    private int id;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    private String nume;
    private String prenume;
    private String parola;
    private String email;

    @Override
    public String toString() {
        return "Voluntar{" +
                " id = "+ this.getId()+" "+
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", parola='" + parola + '\'' +
                ", email='" + email + '\'' +
                '}';
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getParola() {
        return parola;
    }

    public void setParola(String parola) {
        this.parola = parola;
    }


    public Voluntar(String nume, String prenume, String email, String parola) {
        this.nume = nume;
        this.prenume = prenume;
        this.email = email;
        this.parola = parola;
    }



    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Voluntar)) return false;
        Voluntar voluntar = (Voluntar) o;
        return Objects.equals(nume, voluntar.nume) && Objects.equals(prenume, voluntar.prenume) && email.equals(voluntar.email) && Objects.equals(parola, voluntar.parola);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nume, prenume, email, parola);
    }
}

File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Domain/Voluntar.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain
{
    class Voluntar : Entity<int>
    {
        string nume;
        string prenume;
        string email;
        string parola;

        public Voluntar(string nume, string prenume, string email, string parola)
        {
            this.nume = nume;
            this.prenume = prenume;
            this.email = email;
            this.parola = parola;
        }

        public string Nume { get => nume; set => nume = value; }
        public string Prenume { get => prenume; set => prenume = value; }
        public string Email { get => email; set => email = value; }
        public string Parola { get => parola; set => parola = value; }

        public override bool Equals(object obj)
        {
            return obj is Voluntar voluntar &&
                   nume == voluntar.nume &&
                   prenume == voluntar.prenume &&
                   email == voluntar.email &&
                   parola == voluntar.parola;
        }

        public override int GetHashCode()
        {
            int hashCode = 1063577040;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(nume);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(prenume);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(email);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(parola);
            return hashCode;
        }

        public override string ToString()
        {
            return "Id = " + this.Id + " , nume = " + this.GetFullName() + "  ,email = " +
                this.Email + "  ,parola = " + this.Parola;
        }

        public String GetFullName()
        {
            return this.Nume + " " + this.Prenume;

        }
    }
}
